### PR TITLE
blockdev_stream:add nbd support for block stream

### PR DESCRIPTION
**Pull request by aliang123**
Backend:nbd support for block stream

Signed-off-by: Aihua Liang <aliang@redhat.com>
id:2068003

### CODE REVIEW

**Comment by aliang123** [comment]
(01/24) Host_RHEL.m9.u1.nbd.raw.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.x86_64.io-github-autotest-qemu.blockdev_stream_stress.q35: PASS (309.91 s)
(02/24) Host_RHEL.m9.u1.nbd.raw.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.x86_64.io-github-autotest-qemu.blockdev_stream_multiple_blocks.q35: PASS (167.80 s)
(03/24) Host_RHEL.m9.u1.nbd.raw.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.x86_64.io-github-autotest-qemu.blockdev_stream_no_backing.q35: PASS (28.83 s)
(04/24) Host_RHEL.m9.u1.nbd.raw.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.x86_64.io-github-autotest-qemu.blockdev_stream_vm_stop_cont.q35: PASS (292.72 s)
(05/24) Host_RHEL.m9.u1.nbd.raw.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.x86_64.io-github-autotest-qemu.blockdev_stream_cor.q35: PASS (97.82 s)
(06/24) Host_RHEL.m9.u1.nbd.raw.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.x86_64.io-github-autotest-qemu.blockdev_stream_vm_reboot.q35: PASS (298.06 s)
(07/24) Host_RHEL.m9.u1.nbd.raw.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.x86_64.io-github-autotest-qemu.blockdev_stream_on_error_ignore.q35: PASS (82.32 s)
(08/24) Host_RHEL.m9.u1.nbd.raw.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.x86_64.io-github-autotest-qemu.blockdev_stream_powerdown.q35: PASS (102.82 s)
(09/24) Host_RHEL.m9.u1.nbd.raw.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.x86_64.io-github-autotest-qemu.blockdev_stream_speed.q35: PASS (156.35 s)
(10/24) Host_RHEL.m9.u1.nbd.raw.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.x86_64.io-github-autotest-qemu.blockdev_stream_no_space.q35: PASS (32.70 s)
(11/24) Host_RHEL.m9.u1.nbd.raw.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.x86_64.io-github-autotest-qemu.blockdev_stream_cor_base.q35: PASS (38.32 s)
 (12/24) Host_RHEL.m9.u1.nbd.raw.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.x86_64.io-github-autotest-qemu.blockdev_stream_none_existed_overlay.q35: PASS (30.44 s)
(13/24) Host_RHEL.m9.u1.nbd.raw.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.x86_64.io-github-autotest-qemu.blockdev_stream_to_invalid_node.q35: PASS (46.09 s)
(14/24) Host_RHEL.m9.u1.nbd.raw.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.x86_64.io-github-autotest-qemu.blockdev_stream_forbidden_actions.q35: PASS (57.67 s)
(15/24) Host_RHEL.m9.u1.nbd.raw.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.x86_64.io-github-autotest-qemu.blockdev_stream_subchain.q35: PASS (108.37 s)
(16/24) Host_RHEL.m9.u1.nbd.raw.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.x86_64.io-github-autotest-qemu.blockdev_stream_general.auto_dismiss_on.auto_finalize_on.q35: PASS (137.32 s)
(17/24) Host_RHEL.m9.u1.nbd.raw.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.x86_64.io-github-autotest-qemu.blockdev_stream_general.auto_dismiss_on.auto_finalize_off.q35: PASS (139.17 s)
(18/24) Host_RHEL.m9.u1.nbd.raw.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.x86_64.io-github-autotest-qemu.blockdev_stream_general.auto_dismiss_off.auto_finalize_on.q35: PASS (139.87 s)
(19/24) Host_RHEL.m9.u1.nbd.raw.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.x86_64.io-github-autotest-qemu.blockdev_stream_general.auto_dismiss_off.auto_finalize_off.q35: PASS (144.60 s)
(20/24) Host_RHEL.m9.u1.nbd.raw.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.x86_64.io-github-autotest-qemu.blockdev_stream.q35: PASS (151.69 s)
(21/24) Host_RHEL.m9.u1.nbd.raw.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.x86_64.io-github-autotest-qemu.blockdev_stream_base_itself.q35: PASS (84.02 s)
(22/24) Host_RHEL.m9.u1.nbd.raw.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.x86_64.io-github-autotest-qemu.blockdev_stream_hotunplug.q35: PASS (114.75 s)
(23/24) Host_RHEL.m9.u1.nbd.raw.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.x86_64.io-github-autotest-qemu.blockdev_stream_install.q35: PASS (987.97 s)
(24/24) Host_RHEL.m9.u1.nbd.raw.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.x86_64.io-github-autotest-qemu.blockdev_stream_filter_nodename.q35: PASS (55.00 s)

**Comment by aliang123** [comment]
@zhencliu @ngu-niny or @MiriamDeng, please help to review, thanks.

**Comment by aliang123** [comment]
(1/1) Host_RHEL.m9.u1.nbd.raw.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.x86_64.io-github-autotest-qemu.blockdev_stream.q35: PASS (102.31 s)

File: qemu/tests/cfg/blockdev_stream_dirty_bitmap.cfg [hunk 1]
```diff
@@ -32,3 +32,9 @@
     ceph:
         image_format_data1 = raw
         enable_ceph_data1sn = no
+    nbd:
+        image_format_data1 = raw
+        enable_nbd_data1sn = no
+        nbd_port_data1 = 10831
+        remove_image_data1 = yes
+        force_create_image_data1 = yes
```
Comment (zhencliu) [inline qemu/tests/cfg/blockdev_stream_dirty_bitmap.cfg:40]: here, both 'yes' -> 'no' ?

Comment (aliang123) [reply qemu/tests/cfg/blockdev_stream_dirty_bitmap.cfg]: Have updated, thanks zhenchao.

**Comment by zhencliu** [comment]
LGTM

**Comment by MiriamDeng** [comment]
Will provide the test results today.

**Comment by MiriamDeng** [comment]
> Will provide the test results today.

Pasted the test log in the auto bug, and there's several failure test cases, could you help to check them @aliang123 , thanks !

**Comment by aliang123** [comment]
> > Will provide the test results today.
> 
> Pasted the test log in the auto bug, and there's several failure test cases, could you help to check them @aliang123 , thanks !

All failed issue are not related to nbd backend, they related with an auto bug that can't find the correct disk label and houqi is working on it.

**Comment by MiriamDeng** [comment]
> > > Will provide the test results today.
> > 
> > 
> > Pasted the test log in the auto bug, and there's several failure test cases, could you help to check them @aliang123 , thanks !
> 
> All failed issue are not related to nbd backend, they related with an auto bug that can't find the correct disk label and houqi is working on it.

Thanks for pointing it out, LGTM then

**Pull request merged**